### PR TITLE
Fix #135: Integrated toast notifications for actions

### LINKED ISSUE #133: Show Notification when the link to the user story is copied
## Describe the bug

Click on the copy link button, no message is shown to the user whether the link is copied or not. 

## To Reproduce

1. Go to https://userstory.eosdesignsystem.com/story/60e6df9180886200152e34b9

## Expected behavior

A notification message should be shown.

## Logs
If applicable, add logs to help explain your problem (e.g. the error message and/or exception traceback).

https://user-images.githubusercontent.com/44670961/155420328-d4044dd0-1bfc-4f9d-ad2c-2b4a66636ed5.mp4

## Environment (please complete the following information):

- OS: Ubuntu 20.04
- Browser: Firefox 97.1

### LINKED ISSUE #135: [Feature] Add toast notifications for actions
## Is your feature request related to a problem? Please describe.

Currently, in the application, we do not have any toast notifications for the features. When the user performs an important action, like creating a story, or editing it, there is no confirmation message generated about it. 

## Describe the solution you'd like

It would be better to add messages of whether an action was successful or not. This can be done using `react-toastify` or `react-hot-toast` packages.

## Additional context

This would make the application more interactive.

### PR DESCRIPTION
**Pull request by kailash360**
**Issue Number**

fixes #135 
fixes #133 

**Describe the changes you've made**

I have added toast notifications for different functionalities of the application like creating a new story, editing a story, loggin into the application, copying story link to clipboard. To implement the toast notifications, I have added the `react-hot-toast` as a dependency of the application.

**Describe if there is any unusual behavior (Any Warning) of your code(Write `NA` if there isn't)**

NA

**Additional context (OPTIONAL)**

NA

**Test plan (OPTIONAL)**

NA

**Checklist**

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

**Provide a Deployed link of route/page that needs to review**

Preview: Deploy preview link here with the appropriate route

image

image

image

image

image

### CODE REVIEW

File: src/pages/ChangePassword.js [hunk 1]
```diff
@@ -38,7 +38,12 @@ const ChangePassword = () => {
         id: localStorage.getItem('id')
       })
       setResponse(reply)
-    } catch (e) {}
+      console.log(reply)
```
[the remark this answers is not in the document]
Comment (kailash360) [reply src/pages/ChangePassword.js]: @cyntss 

Removed the logging in the latest push

File: src/pages/ChangePassword.js [hunk 2]
```diff
@@ -38,7 +38,11 @@ const ChangePassword = () => {
         id: localStorage.getItem('id')
       })
       setResponse(reply)
-    } catch (e) {}
+      toast.success('Password changed successfully')
+    } catch (e) {
+      console.log(e.message)
```
Comment (kbhutani0001) [inline src/pages/ChangePassword.js:43]: console.error

File: src/pages/Login.js [hunk 3]
```diff
@@ -44,6 +45,7 @@ export const Login = (props) => {
       dispatch({
         type: 'AUTHENTICATE'
       })
+      toast.success(`Logged in successfully as ${payload.user.username}`)
       navigate('/', { replace: true })
     } catch (e) {}
```
Comment (kbhutani0001) [inline src/pages/Login.js:50]: do we need to handle this?

File: src/pages/MyProfile.js [hunk 4]
```diff
@@ -24,9 +24,15 @@ const MyProfile = () => {
...
-    if (response) {
-      setUpdated(true)
+    try {
+      const response = await userStory.updateUser({ id: userId, ...user })
+      if (response) {
+        toast.success('Profile updated successfully')
+        setUpdated(true)
+      }
+    } catch (err) {
+      console.log(err.message)
```
Comment (kbhutani0001) [inline src/pages/MyProfile.js:34]: use console.error everywhere when logging error

File: src/pages/Login.js (hunk 3 above)
Comment (kailash360) [reply src/pages/Login.js:50]: We are calling the `login` service there. So, we need to handle any unexpected error in the the service. Although, there is already a `catch` block in the services itself to handle the errors. Should I remove this try-catch block?

File: src/pages/MyProfile.js (hunk 4 above)
Comment (kailash360) [reply src/pages/MyProfile.js:34]: Alright

File: src/pages/ChangePassword.js (hunk 2 above)
Comment (kailash360) [reply src/pages/ChangePassword.js:43]: Alright

**Comment by kailash360** [comment]
@kbhutani0001 
I have pushed the changes

**Pull request merged**